### ISSUE TITLE
Coadd the targeting columns with BITWISE_OR

### ISSUE THREAD

**Issue by stephjuneau** [desihub/desispec#2077]
Move the targeting columns (all of them: there are 17 including CMX) from `fibermap_target_cols` to instead be combined as part of `desispec.coaddition.coadd_fibermap()` where the combination is to apply a BITWISE_OR.

This will both fix known issues with some SV1 secondaries but also prevent possible future conflicts if some targets were to acquire new secondary or tertiary targeting bits at any point during the full DESI survey duration.

Context added by @sbailey: currently `coadd_fibermap` incorrectly assumes that targeting columns (e.g. DESI_TARGET) are like tractor photometry columns (e.g. FLUX_R) and if a target is observed multiple times, you can use the value from any of the observations and it will be the same in all of the others.  There are corner cases related to secondaries where that is not the case and the targeting bits for a given target can evolve.  We think this only *adds* new bits, but never *removes* a bit, and that a bitwise-OR of any values found is a good enough representation of what the coadded DESI_TARGET etc. should be.

**Comment by geordie666** [comment]
A suggestion: When writing this code, raise an error if any post-SV1 targets have different targeting bits for the same `TARGETID`. 

We understand why a `TARGETID` could have different targeting bits for SV1 (the targeting code was still evolving for the very first observations, and secondary targets were still being introduced as a targeting class). But, if a `TARGETID` has different targeting bits after these initial SV1 observations then I think we'd want to know why, to fix any potential issues with uniqueness, and to document the cases.

**Comment by stephjuneau** [comment]
Hi @geordie666, for the ToO's I had a list of 230 TARGETIDs that needed to be fixed (in the EDR zcat patch file). Those were all in SV3 and the other decision was to set DESI_TARGET=0 for the ToOs and only keep the info in the SV3_* targeting bits. So this may be a different ticket / fix than this one.

**Comment by araichoor** [comment]
@stephjuneau:

I ve not closely followed that issue/thread, but about the 230 sv3 toos, in case it s relevant:

early in sv3,  the SV3_{DESI,SCND}_TARGET values were not propagated for ToOs.
this has been patched in the 2022, Oct. for the fiberassign files (https://github.com/desihub/fiberassign/pull/436).
and this patching must have been picked up in the iron production (but not for in fuji, which was done in early 2021).

**Comment by sbailey** [comment]
@stephjuneau @geordie666 @araichoor - could one of you dig into the email/slack history and post specific examples impacted by this so that we can test that any fix actually addresses the known cases?  Thanks.

**Comment by geordie666** [comment]
I'll volunteer to do this, but I'm heading out for the day so won't get to it until tomorrow morning.

**Comment by araichoor** [comment]
if I m correct, there are lots of examples of targets having different DESI_TARGET (or e.g. SCND_TARGET) columns in the database in the personal slack thread: https://desisurvey.slack.com/archives/C055389PME2/p1682614769834419
@sbailey: you were in it, so I let you check that thread, and post here relevant examples -- as I m not 100% sure of what s under consideration here (or we can briefly chat next Monday, and I can do it).

**Comment by stephjuneau** [comment]
Hi @sbailey the small targeting patch file that we included with the zcat VAC should contain most such cases. It's a small file that can be used to select TARGETIDs to check. See info here:  https://data.desi.lbl.gov/doc/releases/edr/vac/zcat/#healpix-coadded-redshift-catalog-zpix

**Comment by geordie666** [comment]
A condensed review of some cases for spot-checking:

1. Here are the known cases where the `SV1_MWS_TARGET` bit changes from early versions of the `desitarget` code but the `SV1_DESI_TARGET` bit doesn't:

`39628516421013477, 39632940044388381, 39632950194603031`

2. For the Main Survey, where target bits changed between version `1.0.0` and `1.1.1` of the `desitarget` code, here are some `TARGETID`s and the values of `DESI_TARGET` and `SCND_TARGET` that arise from the different versions:

```
TARGETID  DESI_TARGET SCND_TARGET
39627897635344542 [(6917529027641344004, 34359738368)]
39627897635344542 [(2305843009213956100, 0)]
------------------------------------------------------------------
39627909693969839 [(6917529027641344004, 34359738368)]
39627909693969839 [(2305843009213956100, 0)]
------------------------------------------------------------------
39627915696017550 [(2305843009213956100, 0)]
39627915696017550 [(6917529027641344004, 34359738368)]
------------------------------------------------------------------
39627921765176832 [(6917529027641344004, 100663296)]
39627921765176832 [(2305843009213956100, 0)]
------------------------------------------------------------------
39627933790241326 [(2305843009213956100, 0)]
39627933790241326 [(6917529027641344004, 34359738368)]
------------------------------------------------------------------
39627933807022241 [(2305843009213956100, 0)]
39627933807022241 [(6917529027641344004, 34460401664)]
------------------------------------------------------------------
39627945811116839 [(2305843009213956100, 0)]
39627945811116839 [(6917529027641344004, 34359738368)]
------------------------------------------------------------------
39627945836283400 [(6917529027641409541, 34460401664)]
39627945836283400 [(2305843009214021637, 0)]
------------------------------------------------------------------
39628223046226317 [(6917529027641344004, 34359738368)]
39628223046226317 [(2305843009213956100, 0)]
------------------------------------------------------------------
39628234505067881 [(2305843009213956100, 0)]
39628234505067881 [(6917529027641344004, 34359738368)]
------------------------------------------------------------------
39628240213514125 [(6917529027641344004, 100663296)]
39628240213514125 [(2305843009213956100, 0)]
```

3. Here are some TARGETIDs for cases where we went on-sky with version `0.47.0` of the `desitarget` code (_before_ we implemented secondary targets) and then subsequently reobserved targets with a later version of the code (_after_ we included secondary targets). These were from `TILEID`s `80690` and `80691`:

```
80690 [39632940065359576, 39632940065360515, 39632940073748950, 39632940073750247, 39632945140469357, 39632945144663513, 39632950194603290]
80691 [39632929852228688, 39632929852229953, 39632929856423605, 39632929856425410, 39632929860619211, 39632940044388890]
```

I'll try to compile an exhaustive list today so we can check we've addressed all the corner cases. I'll cross-check that against @stephjuneau's patch file.

**Comment by geordie666** [comment]
One additional comment: My item 3 above is all for changes to `SV1_DESI_TARGET`. I think @stephjuneau's patch file is a _fourth_ case, which reflects updates to how ToOs were implemented.

**Comment by geordie666** [comment]
I spooled through `fuji` and confirm that @stephjuneau's patch file (`/global/cfs/cdirs/desi/public/edr/vac/edr/zcat/fuji/v1.0/zall-pix-targeting-patch-edr.fits`) is a good resource to use for finding problematic targets. 

I was incorrect that Stephanie's file just deals with the ToOs, it also covers my Items 1. and 3. from above. For some reason Stephanie's file is missing `39632940044388381` and the cases on tile `80691`. This could be because we were discussing both potential and actual assignments in the Slack thread to which Anand linked. But, regardless, Stephanie's file captures most of the problematic cases I could find.

So, bottom line, I'd use Stephanie's patch file for testing, here.
